src/cpp/polars/Series.cpp

```diff
@@ -311,6 +329,90 @@ namespace polars {
         }
     }
 
+    // TODO: This method needs to be re-factored.
+    polars::Series _window_size_correction(int window_size, bool center, const polars::Series &input){
+        // assume that this function only gets called when window_size > input.size()
+
+        auto n = window_size / 2;
+
+        if(input.size() % 2 == 0){ n = n - 1; };
+
+        // Prepare timestamps
+        auto ts = input.index();
+
+        auto index_0 = ts.at(0);
+        auto index_1 = ts.at(1);
+
+        double delta = std::ceil(std::abs(index_1 -  index_0));
+
+        double new_index_0;
+        arma::vec new_timestamps;
+
+        arma::vec v(n);
+        v.fill(NAN);
+
+        arma::vec single_nan(1);
+        single_nan.fill(NAN);
+
+        arma::vec new_input;
+
+        if( (center == false) && (window_size > input.size() + 1) ) {
+            new_input = arma::join_cols(v, input.values());
+
+            // padding goes on the left
+            new_index_0 = index_0 - n * delta;
+            new_timestamps = arma::linspace(new_index_0, ts.at(ts.size()-1), (n + input.size()));
+
+        } else if(center == false &&  n == 1) {
+
+            new_input =arma::join_cols(single_nan, arma::join_cols(single_nan, input.values()));
+
+            // passing goes at the end
```

Comment:
is this the same as 'padding goes on the right' case in the catch all 'else' below? wonder if they could be combined somehow? but maybe a TODO.

Reply:
Pretty much this method needs to be refactor because there is redundancy for sure, but I'll a dd it as a TODO since want to move on to the next thing.